src/services/refreshToken.ts

```diff
@@ -0,0 +1,68 @@
+import axios, { AxiosError } from 'axios';
+
+import { refreshToken as authenticationRefreshToken } from 'adapters/Authentication';
+import { authenticatedHeader } from 'adapters/BaseAuth';
+import { clearToken, getToken, setToken } from 'helpers/authentication';
+import { DeserializableResponse, deserialize } from 'helpers/deserializer';
+import { paths } from 'routes';
+import { Token } from 'types/token';
+
+let isAlreadyFetchingAccessToken = false;
+let subscribers: { (isFetchSuccess: boolean): void }[] = [];
+
+function onAccessTokenFetched(isFetchSuccess: boolean) {
+  subscribers = subscribers.filter((callback) => callback(isFetchSuccess));
+}
+
+function addSubscriber(callback: { (isFetchSuccess: boolean): void }) {
+  subscribers.push(callback);
+}
+
+const doRefreshToken = (token: string) => {
+  return authenticationRefreshToken(token)
+    .then((response: DeserializableResponse) => {
+      const tokenType = deserialize<Token>(response.data);
+
+      setToken(tokenType);
+      return true;
+    })
+    .catch(() => {
+      clearToken();
+
+      // Redirect to Sign In screen
+      window.location.href = paths.signIn;
+      return false;
+    });
+};
+
+export const refreshToken = (error: AxiosError<unknown>): Promise<unknown> => {
+  const { config: originalRequest } = error;
+  const retryOriginalRequest = new Promise((resolve) => {
+    addSubscriber((isFetchSuccess) => {
+      if (!isFetchSuccess) {
+        resolve({});
+        return;
+      }
+      if (originalRequest) {
+        Object.assign(originalRequest.headers, {
+          ...originalRequest.headers,
+          ...authenticatedHeader(),
+        });
```

Comment:
```suggestion
        Object.assign(originalRequest.headers, {
          ...authenticatedHeader(),
        });
```

Reply:
@nvminhtue nice, thanks 👍

Reply:
Fixed in [de41930](https://github.com/manh-t/react-survey/pull/67/commits/de419303a224598f9b4447ce14e084ddb00e23b9)